decepticonlp/transforms/perturbations.py

```diff
@@ -251,23 +240,19 @@ def apply(self, word: str, **kwargs):
             "z": ["a", "s", "x"],
         }
 
-        # insert typo
-        for pos in positions_to_shift:
-            # no typo insertion for special characters
-            try:
-                typo_list = keys_in_proximity[word[pos]]
-                word[pos] = random.choice(typo_list)
-            except:
-                break
+        for i, c in enumerate(word):
+            # Check Upper
+            cap = c.isupper()
 
-        # reinsert capitalization
-        for i in range(len(word)):
-            if capitalization[i]:
-                word[i] = word[i].upper()
+            # Check if in position and given keys
+            if i in positions_to_shift and c in keys_in_proximity:
+                word[i] = random.choice(keys_in_proximity[c])
+                if cap:
+                    # convert to upper if in upper
+                    word[i] = word[i].upper()
```

Comment:
Add test case to cover this also.
That should resolve the coverage failure?

Reply:
The test case is already there, does code coverage count the number of test cases for logic ?

Reply:
A simpler resolve would be to extend the dictionary to capital letters.

Pros - Defining non alphabetical typos as well, the current implementation and this PR to assume that typos consider letters only, however, I can also put a number in between. So we can define different typos for capital letters

Cons - Will clutter the code.

Should I save this dictionary as a son rather ? This way in future if we wish to add any changes we don't have to mess with the code